File: programando na linguagem java/Converter_Decimal_Binario/src/Converte_Dec_Bin.java
import javax.swing.JOptionPane;
// classe que converte decimal em binario
public class Converte_Dec_Bin {

	private int testa_val_dec() {
		int numDecimal = -1;

		while (numDecimal < 0 || numDecimal > 15) {
			numDecimal = Integer.parseInt(JOptionPane.showInputDialog("Digite um valor entre 0 e 15"));
		}
		return numDecimal;
	}

	private boolean[] valor_em_binario(int valor_decimal) {
		boolean[] binario;
		int valor;

		valor = valor_decimal;

		binario = new boolean[4];

		for (int i = 3; i >= 0; i--) {

			if (valor % 2 == 1) {
				binario[i] = true;
			} else {
				binario[i] = false;
			}

			valor = valor / 2;
		}

		return binario;
	}

	private void exibe_conversao(boolean[] val_binario) {
		JOptionPane.showMessageDialog(null,
				String.valueOf(val_binario[0] + " " + val_binario[1] + " " + val_binario[2] + " " + val_binario[3]));
	}

	private void testa_senha(boolean[] val_binario) {
		if ((!val_binario[0] && val_binario[1]) || (!val_binario[0] && val_binario[2] && val_binario[3])
				|| (val_binario[0] && !val_binario[1] && !val_binario[2])) {
			JOptionPane.showMessageDialog(null, "Acesso permitido!");
		} else {
			JOptionPane.showMessageDialog(null, "Acesso negado!");
		}

	}

	public static void main(String[] args) {
		int sai_prog;
		Converte_Dec_Bin conv;
		boolean[] binario;

		conv = new Converte_Dec_Bin();

		sai_prog = 1;

		while (sai_prog != 0) {

			binario = conv.valor_em_binario(conv.testa_val_dec());

			conv.exibe_conversao(binario);

			conv.testa_senha(binario);

			sai_prog = Integer
					.parseInt(JOptionPane.showInputDialog("Digite 0 para sair ou outro valor para continuar."));

		}
	}
}

File: programando na linguagem python/converte decimal para binario/scr/converte_dec_bin.py
class ConverteDecimalBinario:

    def __init__(self):
        pass

    def testa_valor_decimal(self):
        numero_decimal = int(input('digite um valor entre 0 e 15!\n'))
        while numero_decimal < 0 or numero_decimal > 15:
            numero_decimal = int(input('digite um valor entre 0 e 15!\n'))
        return numero_decimal

    def valor_binario(self, valor_decimal):
        binario = []
        valor = valor_decimal
        for indice in range(4):
            if valor % 2 == 1:
                binario.append(True)
            else:
                binario.append(False)
            valor = valor // 2
        binario.reverse()
        return binario

    def exibe_conversao(self, val_binario):
        print(val_binario)

    def testa_senha(self, val_binario):
        if (not valor_binario[0] and valor_binario[1]) or (
                not valor_binario[0] and valor_binario[2] and valor_binario[3]) or (
                valor_binario[0] and not valor_binario[1] and not valor_binario[2]):
            print('Acesso permitido!')
        else:
            print('Acesso negado!')


if __name__ == '__main__':
    testa_entrada = ConverteDecimalBinario
    fim_laco = -1
    while fim_laco != 0:
        val_decimal = ConverteDecimalBinario.testa_valor_decimal(testa_entrada)
        valor_binario = ConverteDecimalBinario.valor_binario(testa_entrada, val_decimal)
        ConverteDecimalBinario.exibe_conversao(testa_entrada, valor_binario)
        ConverteDecimalBinario.testa_senha(testa_entrada, valor_binario)
        fim_laco = int(input('digite 0 para sair ou outro valor para continuar.\n'))
